examples/event_loop/src/event_loop_example.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+#include <klepsydra/core/publisher.h>
+#include <klepsydra/core/subscriber.h>
+
+#include "spdlog/spdlog.h"
+int main(int argc, char *argv[]) {
+
+  
+  spdlog::set_level(spdlog::level::debug);
+  const int width = 500, height = 500;
+  kpsr::vision_ocv::ImageDataFactory imageCreator(width, height, cv::IMREAD_REDUCED_GRAYSCALE_2, "camera_link");
+  kpsr::high_performance::EventLoopMiddlewareProvider<16> eventLoopProvider(nullptr);
+  const int totalSubscribers = 1, readTimes=6;
+  int totalImagesListened = 0;
+  std::vector<kpsr::Subscriber<kpsr::vision_ocv::ImageData>*> subscribers;
+  std::vector<kpsr::Publisher<kpsr::vision_ocv::ImageData>*> publishers;
+  std::vector<std::unique_ptr<kpsr::vision_ocv::FileImageStreamingService>> readFromDiskServices;
+  std::vector<std::thread> threads;
+
+  eventLoopProvider.start();
+  
+  std::string prefixChannel = "channel_";
+  // creation
+  for (int i = 0; i < totalSubscribers; i++) {
+    std::string channelName = prefixChannel+std::to_string(i);
+    subscribers.push_back(eventLoopProvider.getSubscriber<kpsr::vision_ocv::ImageData>(channelName));
+    subscribers[i]->registerListener(channelName, [&totalImagesListened](const kpsr::vision_ocv::ImageData &event) {
+            std::string imgName = "/home/mario/klepsydra/kpsr-vision-ocv-tutorial/examples/event_loop/out/img" + std::to_string(totalImagesListened++) + ".jpg";
+            cv::Mat outputImage;
+            cvtColor(event.img, outputImage,cv::COLOR_BGR2GRAY);
+            bool statusWrite = cv::imwrite(imgName, outputImage);
+            
+            spdlog::debug("Status imwrite: {}.", statusWrite);
+            spdlog::debug("Eventloop (subscriber) thread ID: {}");
+            
+        });
+    publishers.push_back(eventLoopProvider.getPublisher<kpsr::vision_ocv::ImageData>(channelName,
```

Comment:
Separate creation of subscribers, publishers and services in separate loops. It does not change the code, but it will look clearer.